modules/objdetect/include/opencv2/objdetect/aruco_detector.hpp

```diff
@@ -388,6 +388,67 @@ CV_EXPORTS_W void drawDetectedMarkers(InputOutputArray image, InputArrayOfArrays
+    Mat distCoeffs; // distCoeffs optional vector of distortion coefficients
+    int minMarkers; // minMarkers number of adjacent markers that must be detected to return a charuco corner, default = 3
+    bool tryRefineMarkers; // try to use refine board
+};
+
+class CV_EXPORTS_W CharucoDetector : public ArucoDetector {
+public:
+    /** @brief Basic CharucoDetector constructor
+     *
+     * @param board ChAruco board
+     * @param detectorParams marker detection parameters
+     * @param refineParams marker refine detection parameters
+     * @param charucoParams charuco detection parameters
+     */
+    CV_WRAP CharucoDetector(const Ptr<CharucoBoard> &board,
+                            const RefineCharucoParameters& refineCharucoParameters = RefineCharucoParameters(),
+                            const DetectorParameters &detectorParams = DetectorParameters(),
+                            const RefineParameters& refineParams = RefineParameters());
+
+    CV_WRAP const Ptr<CharucoBoard>& getBoard() const;
+    CV_WRAP void setBoard(const Ptr<CharucoBoard>& board);
+
+    CV_WRAP const RefineCharucoParameters& getCharucoParameters() const;
+    CV_WRAP void setCharucoParameters(RefineCharucoParameters& charucoParameters);
+
+    /**
+     * @brief If nedeed detect aruco markers and interpolate position of ChArUco board corners
+     * @param markerCorners vector of already detected markers corners. For each marker, its four
+     * corners are provided, (e.g std::vector<std::vector<cv::Point2f> > ). For N detected markers, the
+     * dimensions of this array should be Nx4. The order of the corners should be clockwise.
+     * If markerCorners and markerCorners are empty, the detectMarkers() will run and detect aruco markers and ids
+     * @param markerIds list of identifiers for each marker in corners.
+     * If markerCorners and markerCorners are empty, the detectMarkers() will run and detect aruco markers and ids
+     * @param image input image necesary for corner refinement. Note that markers are not detected and
+     * should be sent in corners and ids parameters.
+     * @param board layout of ChArUco board.
+     * @param charucoCorners interpolated chessboard corners
+     * @param charucoIds interpolated chessboard corners identifiers
+     *
+     * This function receives the detected markers and returns the 2D position of the chessboard corners
+     * from a ChArUco board using the detected Aruco markers.
+     *
+     * If markerCorners and markerCorners are empty, the detectMarkers() will run and detect aruco markers and ids
+     *
+     * If camera parameters are provided, the process is based in an approximated pose estimation, else it is based on local homography.
+     * Only visible corners are returned. For each corner, its corresponding identifier is
+     * also returned in charucoIds.
+     */
+    CV_WRAP void detectCharuco(InputArray image, InputOutputArrayOfArrays markerCorners, InputOutputArray markerIds,
```

Comment:
I propose to name it simply "detect" or "detectBoard" like "detectMarkers" for Aruco.

Reply:
rename `detectCharuco` to `detectBoard`